Made `_weighted_percentile` raise value error when weights are all zero. (#31041)
scikit-learn/scikit-learn: doc/whats_new/upcoming_changes/sklearn.dummy/31041.fix.rst

> - In :mod:`dummy` made it so that if you use the median strategy with all zero weights it will raise a value error.

Reviewer: I don't think this is true, `np.median` is used in this case.

> - In :mod:`dummy` made it so that if you use the quantile strategy with all zero weights it will raise a value error.

Reviewer: ```suggestion
- Updated :mod:`dummy` to raise a value error if `strategy="quantile"` and `fit` with
  `sample_weight` being all zeros.
```
